### PR TITLE
docs(framework) Add quickstart examples Docker compose guide

### PR DESCRIPTION
**Pull request by Robert-Steiner**
### Description

-  Adds a guide that explains how to run quickstart examples using docker compose
- depends on #4187  

### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.ai/docs/writing-documentation.html)
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### CODE REVIEW

File: doc/source/docker/run-quickstart-examples-docker-compose.rst [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+   .. code-block:: bash
+
+      $ cd flower/examples/quickstart-sklearn-tabular
+
+#. Append the following lines to the end of the ``pyproject.toml`` file and save it:
+
+   .. code-block:: toml
+      :caption: pyproject.toml
+
+      [tool.flwr.federations.docker-compose]
```
Comment (Robert-Steiner) [inline doc/source/docker/run-quickstart-examples-docker-compose.rst:80]: We could add this to the `pyproject.toml` file to remove this step. wdyt?

File: doc/source/docker/run-quickstart-examples-docker-compose.rst [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+   * - quickstart-mlcube
+     - The example has not yet been updated to work with the latest ``flwr`` version.
+   * - quickstart-mlx
+     - `Requires to run on macOS with Apple Silicon <https://ml-explore.github.io/mlx/build/html/install.html#python-installation>`_.
+   * - quickstart-monai
+     - None
+   * - quickstart-pandas
+     - The example has not yet been updated to work with the latest ``flwr`` version.
+   * - quickstart-pytorch-lightning
+     - Requires an older pip version that is not supported by the Flower Docker images.
```
Comment (Robert-Steiner) [inline doc/source/docker/run-quickstart-examples-docker-compose.rst:167]: If we update the dependencies (pytorch) we can make the quickstart work with docker compose

File: doc/source/docker/run-quickstart-examples-docker-compose.rst [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+     - Limitation/Requirements
+   * - quickstart-fastai
+     - None
+   * - examples/quickstart-huggingface
+     - For CPU-only environments, it requires at least 32GB of memory.
+   * - quickstart-jax
+     - The example has not yet been updated to work with the latest ``flwr`` version.
+   * - quickstart-mlcube
+     - The example has not yet been updated to work with the latest ``flwr`` version.
+   * - quickstart-mlx
```
Comment (Robert-Steiner) [inline doc/source/docker/run-quickstart-examples-docker-compose.rst:160]: We should remove the mentioning of Docker here: https://github.com/adap/flower/blob/541064b79e4de3d95be9ec270ff66da201feb3b6/examples/quickstart-mlx/README.md?plain=1#L70

Comment (jafermarq) [reply doc/source/docker/run-quickstart-examples-docker-compose.rst:99]: https://github.com/adap/flower/pull/4202

File: doc/source/docker/run-quickstart-examples-docker-compose.rst (hunk 1 above)
Comment (jafermarq) [reply doc/source/docker/run-quickstart-examples-docker-compose.rst:53]: I think leaving it for people to do could be a simple but useful learning experience for future projects?

File: doc/source/docker/run-quickstart-examples-docker-compose.rst [hunk 4]
```diff
@@ -0,0 +1,114 @@
+Run Flower Quickstart Examples with Docker Compose
+==================================================
+
+Flower provides a set of `quickstart examples <https://github.com/adap/flower/tree/main/examples>`_
+to help you get started with the framework.
+
+These examples are designed to demonstrate the capabilities of Flower and can be run
+using Docker Compose.
```
Comment (jafermarq) [inline doc/source/docker/run-quickstart-examples-docker-compose.rst:8]:
```suggestion
These examples are designed to demonstrate the capabilities of Flower and by default run using the Simulation Engine. This guide demonstrates how to run them using Flower's Deployment Engine via through Docker Compose.
```

Comment (jafermarq) [reply doc/source/docker/run-quickstart-examples-docker-compose.rst:8]: I thought of rewriting it slightly. Probably we can iterate more. Also, should this be a different paragraph from the first sentence in the doc page? or should we join them ?

File: doc/source/docker/run-quickstart-examples-docker-compose.rst [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+Run the Quickstart Example
+--------------------------
+
+#. Clone the quickstart example you like to run. For example, ``quickstart-sklearn-tabular``:
+
+   .. code-block:: bash
+
+      $ git clone --depth=1 https://github.com/adap/flower.git \
+           && mv flower/examples/quickstart-sklearn-tabular . \
+           && rm -rf flower && cd quickstart-sklearn-tabular
```
Comment (jafermarq) [inline doc/source/docker/run-quickstart-examples-docker-compose.rst:33]:
```suggestion
           && mv flower/examples/quickstart-pytorch . \
           && rm -rf flower && cd quickstart-pythorc
```

Comment (jafermarq) [reply doc/source/docker/run-quickstart-examples-docker-compose.rst:33]: i think the pytorch example or the tf would offer a better experience in general

File: doc/source/docker/run-quickstart-examples-docker-compose.rst [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+      $ docker compose up --build -d
+
+#. Append the following lines to the end of the ``pyproject.toml`` file and save it:
+
+   .. code-block:: toml
+      :caption: pyproject.toml
+
+      [tool.flwr.federations.docker-compose]
+      address  =  "127.0.0.1:9093"
+      insecure  = true
```
Comment (jafermarq) [inline doc/source/docker/run-quickstart-examples-docker-compose.rst:55]:
```suggestion
      address = "127.0.0.1:9093"
      insecure = true
```

File: doc/source/docker/run-quickstart-examples-docker-compose.rst [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+   .. code-block:: bash
+
+      $ docker compose up --build -d
+
+#. Append the following lines to the end of the ``pyproject.toml`` file and save it:
+
+   .. code-block:: toml
+      :caption: pyproject.toml
+
+      [tool.flwr.federations.docker-compose]
```
Comment (jafermarq) [inline doc/source/docker/run-quickstart-examples-docker-compose.rst:53]: I'm thinking if we should name the federation to something different that doesn't use the `docker` or `	compose` keywords. The alternative would be to add a comment saying that the string that comes after `federations.` is up to the user to define (then use is accordingly when doing `flwr run .`

File: doc/source/docker/run-quickstart-examples-docker-compose.rst (hunk 4 above)
Comment (Robert-Steiner) [reply doc/source/docker/run-quickstart-examples-docker-compose.rst]: > I thought of rewriting it slightly. P
That's sounds better! 
I think we can join them. I have no strong preference.

File: doc/source/docker/run-quickstart-examples-docker-compose.rst (hunk 7 above)
Comment (Robert-Steiner) [reply doc/source/docker/run-quickstart-examples-docker-compose.rst:52]: I'm open to alternative name suggestions. we can also call it `local` or `local-insecure` 

I'll add a comment, but this i feel this information belongs more to the flwr cli docs.

File: doc/source/docker/run-quickstart-examples-docker-compose.rst (hunk 1 above)
Comment (Robert-Steiner) [reply doc/source/docker/run-quickstart-examples-docker-compose.rst:52]: I think it is ok to do it for a single example but i personally got a little bit annoyed when i tried different examples.

File: doc/source/docker/run-quickstart-examples-docker-compose.rst (hunk 7 above)
Comment (jafermarq) [reply doc/source/docker/run-quickstart-examples-docker-compose.rst:52]: how about `local-deployment`? I agree info should better be left to other parts of the docs.

Comment (Robert-Steiner) [reply doc/source/docker/run-quickstart-examples-docker-compose.rst]: `local-deployment` sounds good to me!

**Review by jafermarq — DISMISSED** [review]

File: doc/source/docker/run-quickstart-examples-docker-compose.rst [hunk 8]
```diff
@@ -0,0 +1,122 @@
+Run Flower Quickstart Examples with Docker Compose
+==================================================
+
+Flower provides a set of `quickstart examples <https://github.com/adap/flower/tree/main/examples>`_
+to help you get started with the framework. These examples are designed to demonstrate the
+capabilities of Flower and by default run using the Simulation Engine. This guide demonstrates
+how to run them using Flower's Deployment Engine via through Docker Compose.
```
Comment (danieljanes) [inline doc/source/docker/run-quickstart-examples-docker-compose.rst:7]:
```suggestion
how to run them using Flower's Deployment Engine via Docker Compose.
```

**Review by danieljanes — APPROVED** [review]

**Pull request merged**